### ISSUE TITLE
Dsdv should not use "strstr" to find wireless interface 

### ISSUE THREAD

**Issue by edlongman** [inet-framework/inet#689]
https://github.com/inet-framework/inet/blob/0aacd06e8b300518ab3adb7e9c7e0c5daa436fea/src/inet/routing/dsdv/Dsdv.cc#L74-L101

Using strstr seems like a horrible way to test if it is a Ieee80211 interface. Can the `ie->isWireless()` be used instead. Does this need to be used in conjuction with `!ie->isLoopback()`

**Comment by levy** [comment]
Yes, indeed. It's not even true BTW, having wlan in the interface name doesn't guarantee that it's an IEEE 802.11 interface. Regarding skipping loopback interface, that makes sense too.

**Comment by aarizaq** [comment]
I think that the interface prefix should be obtained like parameter.

This is the solution that I use with Aodv-uu, Olsr ....

string interfaces = default("prefix(wlan)");

     struct InterfaceIdentification
    {
        NetworkInterface *interfacePtr = nullptr;
        int index = -1;
    };

    const char *interfaces = par("interfaces");
    cStringTokenizer tokenizerInterfaces(interfaces);
    while ((token = tokenizerInterfaces.nextToken()) != nullptr) {
            if ((prefixName = strstr(token, "prefix")) != nullptr) {
                const char *leftparenp = strchr(prefixName, '(');
                const char *rightparenp = strchr(prefixName, ')');
                std::string interfacePrefix;
                interfacePrefix.assign(leftparenp + 1, rightparenp - leftparenp - 1);
                for (int i = 0; i < ift->getNumInterfaces(); i++) {
                    ie = ift->getInterface(i);
                    name = ie->getInterfaceName();
                    if ((strstr(name, interfacePrefix.c_str()) != nullptr) {
                        InterfaceIdentification interface;
                        interface.interfacePtr = ie;
                        interface.index = i;
                        num_80211++;
                        interfaceVector->push_back(interface);
                    }
                }
            }

Using this solution, you can easily to configure the interfaces that you want to use (wired or wireless)
DSDV only uses an interface, but it could use several interfaces, even it can be used in a wired network, in the end is a protocol quite similar to RIP
